src/algebra/star/pi.lean

```diff
@@ -44,4 +44,18 @@ instance {R : Type w}
   star_module R (Π i, f i) :=
 { star_smul := λ r x, funext $ λ i, star_smul r (x i) }
 
+lemma star_update [Π i, has_star (f i)] [decidable_eq I]
+  (h : Π (i : I), f i) (i : I) (a : f i) :
+  star (function.update h i a) = function.update (star h) i (star a) :=
+begin
```

Comment:
You should be able to golf this to use the same proof as

https://github.com/leanprover-community/mathlib/blob/876cb64e1d1bd6289e2f71a0fb69dea0005eb3dc/src/algebra/group/pi.lean#L350

This also suggests the lemma should be reversed and called `function.update_star`